src/hooks/shards/useShardsList.ts

```diff
@@ -1,85 +1,80 @@
 import { useUserStore } from 'context/User/useUserContextStore';
 import { ShardClient, ShardSelector } from 'data-plane-gateway';
-import useGatewayAuthToken from 'hooks/gatewayAuthToken/useGatewayAuthToken';
+import { Shard } from 'data-plane-gateway/types/shard_client';
 import { useMemo } from 'react';
 import { logRocketConsole } from 'services/shared';
 import useSWR from 'swr';
-import {
-    dataPlaneFetcher_list,
-    shouldRefreshToken,
-} from 'utils/dataPlane-utils';
+import { authorizeTask, dataPlaneFetcher_list } from 'utils/dataPlane-utils';
 
 // These status do not change often so checking every 30 seconds is probably enough
 const INTERVAL = 30000;
 
 const useShardsList = (catalogNames: string[]) => {
     const session = useUserStore((state) => state.session);
 
-    const { data: gatewayConfig, refresh: refreshAccess } =
-        useGatewayAuthToken(catalogNames);
-
-    const shardClient = useMemo(() => {
-        if (gatewayConfig?.gateway_url && gatewayConfig.token) {
-            const authToken = gatewayConfig.token;
-            const baseUrl = new URL(gatewayConfig.gateway_url);
-
-            return new ShardClient(baseUrl, authToken);
-        } else {
-            return null;
-        }
-    }, [gatewayConfig]);
-
-    const taskSelector = new ShardSelector();
-    catalogNames.forEach((name) => taskSelector.task(name));
-
     const fetcher = async (_url: string) => {
         // We check this in the swrKey memo so this should never actually happen
-        if (!(shardClient && session)) {
+        if (!session) {
             return { shards: [] };
         }
 
-        const dataPlaneListResponse = await dataPlaneFetcher_list(
-            shardClient,
-            taskSelector,
-            'ShardsList'
-        );
+        const shardPromises = catalogNames.map(async (name) => {
```

Comment:
I think we need to limit this and make sure there is some limits of how many will be called all at once.